server/src/Api/Controllers/CoursesController.cs

```diff
@@ -44,15 +46,15 @@ public async Task<IActionResult> GetAllCourses()
     public async Task<IActionResult> UpdateCourse(int courseId, [FromBody] UpdateCourseDto request)
     {
         request.ThrowIfRequestArgumentNull(nameof(request));
```

Comment:
We no longer need the null exception, can handle it using results

```suggestion
        return await Result
            .FailIf(request is null, {some error})
            .ToResult<UpdateCourseDto>()
            .Bind(dto => _courseService.UpdateCourse(courseId, dto))
            .ToActionResult();
```